util/process_windows.go

```diff
@@ -19,23 +19,88 @@
 package util
 
 import (
+	"errors"
 	"os"
+	"syscall"
 	"unsafe"
 
 	"golang.org/x/sys/windows"
 )
 
-// KillProcess kills a running OS process
-func KillProcess(pid int, _ os.Signal) error {
+const (
+	ERROR_INVALID_PARAMETER = syscall.Errno(87)
+
+	processTerminateWaitInMs = 1000
+)
+
+// FindProcess looks for a running process by its pid
+func FindProcess(pid int) (*os.Process, error) {
+	var h syscall.Handle
+
+	process, err := os.FindProcess(pid)
+	if err != nil {
+		// NOTE: Unlike Unix, Windows tries to open the target process in order to kill it.
+		//       ERROR_INVALID_PARAMETER is returned if the process does not exists.
+		//       To mimic other OS behavior, if the process does not exist, don't return an error
+		var syscallError syscall.Errno
+
+		if errors.As(err, &syscallError) {
+			if syscallError == ERROR_INVALID_PARAMETER {
+				return nil, nil
+			}
+		}
+		return nil, err
+	}
 
-	p, err := os.FindProcess(pid)
+	// If we have a process, check if it is terminated
+	h, err = syscall.OpenProcess(syscall.SYNCHRONIZE, false, uint32(pid))
 	if err == nil {
+		defer func() {
+			_ = syscall.CloseHandle(h)
+		}()
 
-		for _, v := range getChildrenProcesses(pid) {
-			_ = v.Kill()
+		ret, e2 := syscall.WaitForSingleObject(h, 0)
+		if e2 == nil && ret == syscall.WAIT_OBJECT_0 {
+			return nil, nil
 		}
+	}
 
-		err = p.Kill()
+	return process, nil
+}
+
+// KillProcess kills a running OS process
+func KillProcess(pid int, signal os.Signal) error {
```

Comment:
Your `KillProcess` implementation could be a bit better:
* When killing processes, consider that there could be a deep nested tree. Your implementation currently only handle the first layer of child processes.
* When killing processes, you'd better try to terminate them all at once, without blocking.
* Last, after you killed all the child processes, you should reiterate on the child processes to ensure no new children were spawn during the function execution.